common/dynamicconfig/config.go

```diff
@@ -171,53 +171,51 @@ type IntPropertyFnWithWorkflowTypeFilter func(domainName string, workflowType st
 type DurationPropertyFnWithWorkflowTypeFilter func(domainName string, workflowType string) time.Duration
 
 // GetProperty gets a interface property and returns defaultValue if property is not found
-func (c *Collection) GetProperty(key Key, defaultValue interface{}) PropertyFn {
+func (c *Collection) GetProperty(key Key) PropertyFn {
 	return func() interface{} {
-		val, err := c.client.GetValue(key, defaultValue)
+		val, err := c.client.GetValue(key)
 		if err != nil {
 			c.logError(key, nil, err)
 		}
-		c.logValue(key, nil, val, defaultValue, reflect.DeepEqual)
+		c.logValue(key, nil, val, key.DefaultValue(), reflect.DeepEqual)
 		return val
 	}
 }
 
 // GetIntProperty gets property and asserts that it's an integer
-func (c *Collection) GetIntProperty(key Key, defaultValue int) IntPropertyFn {
+func (c *Collection) GetIntProperty(key IntKey) IntPropertyFn {
 	return func(opts ...FilterOption) int {
 		filters := c.toFilterMap(opts...)
 		val, err := c.client.GetIntValue(
 			key,
 			filters,
-			defaultValue,
```

Comment:
pretty sure this is necessary - without it I believe our dynamic config client will return zero values if we haven't overridden / defined the key, rather than the configured default value.

Reply:
I think it's something we want. We can prevent from using undefined keys.

Reply:
(from IRL chat) ah, yeah, I'm getting confused by remembering the old API.  the key contains default now, so there's no need to pass it a second time.

👍